pkg/scheduler/eventhandlers.go

```diff
@@ -148,6 +148,10 @@ func (sched *Scheduler) addPodToSchedulingQueue(obj interface{}) {
 	pod := obj.(*v1.Pod)
 	logger.V(3).Info("Add event for unscheduled pod", "pod", klog.KObj(pod))
 	sched.SchedulingQueue.Add(logger, pod)
+
+	sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(logger, framework.UnscheduledPodAdd, nil, pod, func(p *v1.Pod) bool {
+		return p.UID != pod.UID
```

Comment:
`p.UID != pod.UID` preCheck won't be needed if we run `MoveAllToActiveOrBackoffQueue` before `SchedulingQueue.Add()`